src/config.rs

```diff
@@ -1,81 +1,67 @@
+#[serde(default)]
 pub struct Config {
-    #[serde(default = "default_player_command")]
     pub player_command: String,
-
-    #[serde(default = "default_player_args")]
     pub player_args: Vec<String>,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Self {
-            player_command: default_player_command(),
-            player_args: default_player_args(),
+            player_command: "mpv".to_owned(),
+            player_args: vec![],
         }
     }
 }
 
 impl Config {
+    pub const FILENAME: &str = "ff2mpv-rust.json";
+
     pub fn build() -> Self {
-        if let Ok(config) = Config::parse_config_file() {
-            config
-        } else {
-            Config::default()
-        }
+        Config::parse_config_file().unwrap_or_default()
     }
 
     pub fn parse_config_file() -> Result<Self, FF2MpvError> {
-        let config_path = Config::get_config_location();
-        if !config_path.exists() {
-            return Err(FF2MpvError::NoConfig);
-        }
-
+        let config_path = Config::get_config_location()?;
         let string = fs::read_to_string(config_path)?;
         let config = serde_json::from_str(&string)?;
 
         Ok(config)
     }
 
-    #[cfg(target_family = "unix")]
-    fn get_config_location() -> PathBuf {
-        let mut path = PathBuf::new();
-
-        if let Ok(home) = env::var("XDG_CONFIG_HOME") {
-            path.push(home);
-        } else if let Ok(home) = env::var("HOME") {
-            path.push(home);
-            path.push(".config");
+    /// Returns a *valid* config path, i.e. it exists and is readable by the
+    /// current user. If the config path is not valid, an [`FF2MpvError`] is
+    /// returned.
+    fn get_config_location() -> Result<PathBuf, FF2MpvError> {
+        let config_dir = if cfg!(target_family = "unix") {
+            if let Ok(home) = env::var("XDG_CONFIG_HOME") {
+                PathBuf::from(home)
+            } else if let Ok(home) = env::var("HOME") {
+                PathBuf::from(home).join(".config")
+            } else {
+                PathBuf::from("/etc")
+            }
+        } else if cfg!(target_family = "windows") {
+            env::var("APPDATA")
+                .map_err(|_| FF2MpvError::NoConfig)?
+                .into()
         } else {
-            path.push("/etc");
-        }
-
-        path.push("ff2mpv-rust.json");
-        path
-    }
+            unimplemented!("This platform is not supported")
+        };
 
-    #[cfg(target_family = "windows")]
-    fn get_config_location() -> PathBuf {
-        let mut path = PathBuf::new();
-        let appdata = env::var("APPDATA").unwrap();
-
-        path.push(appdata);
-        path.push("ff2mpv-rust.json");
-        path
+        let path = config_dir.join(Config::FILENAME);
+        match path.try_exists() {
+            Ok(true) => Ok(path),
+            // Broken symbolic link to config file.
+            Ok(false) => Err(FF2MpvError::NoConfig),
```

Comment:
Why would we use `try_exists` instead of `exists`?  This was designed to fallback to default config in case of an error.
See [`Path.exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists) in `std` docs.
> This is a convenience function that coerces errors to false